packages/components/src/Typography.astro

```diff
@@ -0,0 +1,118 @@
+---
+interface Props {
+  variant:
+    | "h1"
+    | "h2"
+    | "h3"
+    | "h4"
+    | "h5"
+    | "h6"
+    | "subtitle1"
+    | "subtitle2"
+    | "body1"
+    | "body2"
+    | "caption"
+    | "overline";
+  override?:
+    | "h1"
+    | "h2"
+    | "h3"
+    | "h4"
+    | "h5"
+    | "h6"
+    | "subtitle1"
+    | "subtitle2"
+    | "body1"
+    | "body2"
+    | "caption"
+    | "overline";
```

Comment:
```suggestion
  as?:
    | "h1"
    | "h2"
    | "h3"
    | "h4"
    | "h5"
    | "h6"
    | "p"
    | "span";
```

I would suggest to conform to MUI here and just accept the actual element. If they want to override it, it's likely they care what the underlying HTML element is for semantic reasons.